src/coreComponents/constitutive/fluid/singlefluid/SingleFluidBase.hpp

```diff
@@ -276,6 +276,14 @@ class SingleFluidBase : public ConstitutiveBase
   virtual real64 defaultDensity() const = 0;
   virtual real64 defaultViscosity() const = 0;
 
+/**
+ * @brief Get the thermal flag.
+ * @return boolean value indicating whether the model can be used to assemble the energy balance equation or not
+ * @detail if isThermal is true, the constitutive model compute the enthalpy and internal energy of the phase.
+ *         This can be used to check the compatibility of the constitutive model with the solver
+ */
+  virtual bool isThermal() const { return false; }
```

Comment:
do you need to add one with `true` in `ThermalCompressibleSinglePhaseFluid `?

Reply:
@tjb-ltk Is this resolved?